python/robot_properties_solo/config.py

```diff
@@ -55,25 +91,70 @@ class SoloConfig:
     # joints
     nb_joints = robot_model.nv - 6
 
-    # control time period
-    control_period = 0.001
-    dt = control_period
+    # mapping between the ctrl vector in the device and the urdf indexes
+    urdf_to_dgm = (0,1,2,3,4,5,6,7)
 
+    map_joint_name_to_id = {}
+    map_joint_limits = {}
+    for i, (name, lb, ub) in enumerate(zip(robot_model.names[1:],
+                                       robot_model.lowerPositionLimit,
+                                       robot_model.upperPositionLimit)):
+        map_joint_name_to_id[name] = i
+        map_joint_limits[i] = [float(lb), float(ub)]
 
-    # maxCurrent = 12 # Ampers
-    max_current = 2
+    # Define the initial state.
+    initial_configuration = [0.2,0,0.4, 0,0,0,1] + 4*[0.8,-1.6]
+    initial_velocity = (8+6)*[0,]
 
-    # maximum torques
-    max_torque = motor_torque_constant * max_current
+    q0 = zero(robot_model.nq)
+    q0[:] = np.asmatrix(initial_configuration).T
+    v0 = zero(robot_model.nv)
+    a0 = zero(robot_model.nv)
 
-    # maximum control one can send, here the control is the current.
-    max_control = max_current
 
-    # mapping between the ctrl vector in the device and the urdf indexes
-    urdf_to_dgm = (0,1,2,3,4,5,6,7)
 
-    # ctrl_manager_current_to_control_gain I am not sure what it does so 1.0.
-    ctrl_manager_current_to_control_gain=1.0
+class Solo12Config(SoloAbstract):
+    # name that is used by every other entities. I do not exactly why...
+    robot_family = "solo"
+    robot_name = "solo12"
+
+    # here we use the same urdf as for the quadruped but without the freeflyer
+    urdf_path = (
+        join(rospkg.RosPack().get_path("robot_properties_" + robot_family),
+             "urdf",
+             robot_name + ".urdf")
+    )
+
+    meshes_path = [
+      dirname(rospkg.RosPack().get_path("robot_properties_" + robot_family))
+    ]
+
+    yaml_path = (
+        join(rospkg.RosPack().get_path("robot_properties_" + robot_family),
+             "config",
+             "dgm_parameters.yaml")
+    )
+
+    # The inertia of a single blmc_motor
+    motor_inertia = 0.0000045
```

Comment:
Why adding this again in Solo12?
idem for the gear ratio...
Alle these data are the same no?

Reply:
The name is defined but it cannot be used during the initialization of the class.

Here is an example [1]: 

```
class A(object):
  foo = 1
  bar = foo

class B(A):
  baz = foo

B()
> NameError: name 'foo' is not defined
```

Therefore, I've ended up copying some of the definitions.

Do you see a way to have the class variables and avoid the duplications? I was thinking about changing the class variables to instance variables (assign them in an `__init__` call). This would then change the way we use the class right now (e.g. a `SoloConfig.motor_inertia` would become `config = SoloConfig(); config.motor_inertia`.

[1]: https://colab.research.google.com/drive/1uOFFt5lPYsnQRJyTzNryGOlV0dszlaWP